Polygon_mesh_processing/test/Polygon_mesh_processing/remeshing_quality_test.cpp

```diff
@@ -0,0 +1,110 @@
+  PMP::Adaptive_sizing_field sizing_field(tol, edge_min_max, faces(mesh), mesh);
+  const unsigned int nb_iter = 5;
+
+  PMP::isotropic_remeshing(
+      faces(mesh),
+      sizing_field,
+      mesh,
+      PMP::parameters::number_of_iterations(nb_iter).number_of_relaxation_steps(3)
+      );
+
+  /*
+   * More information on quality metrics can be found here: https://ieeexplore.ieee.org/document/9167456
+   */
+  std::cout << "Remeshing done, checking triangle quality...\n" << std::endl;
+  double qmin = (std::numeric_limits<double>::max)();      // minimum triangle quality
+  double qavg = 0.;           // average quality
+  double min_angle = (std::numeric_limits<double>::max)(); // minimum angle
+  double avg_min_angle = 0.; // average minimum angle
+  for (auto face : mesh.faces())
+  {
+    if (PMP::is_degenerate_triangle_face(face, mesh))
+    {
+      std::cout << "Found degenerate triangle!" << std::endl;
+      continue;
+    }
+
+    // Calculate Q(t) triangle quality indicator
+    std::vector<Mesh::Point> pts; pts.reserve(3);
+    for (auto vx : mesh.vertices_around_face(mesh.halfedge(face)))
+      pts.push_back(mesh.point(vx));
+
+    double half_perim = 0.; // half-perimeter
+    double lmax = 0.;       // longest edge
+    for (int i = 0; i < 3; ++i)
+    {
+      const double length = CGAL::sqrt(CGAL::squared_distance(pts[i], pts[(i + 1) % 2]));
+
+      half_perim += length;
+      if (length > lmax) lmax = length;
+    }
+    half_perim /= 2.;
+    const double area = CGAL::sqrt(CGAL::squared_area(pts[0], pts[1], pts[2]));
+
+    const double face_quality = 6. / CGAL::sqrt(3.) * area / half_perim / lmax;
+
+    qavg += face_quality;
+    if (face_quality < qmin) qmin = face_quality;
+
+    // Calculate minimum angle
+    const auto v0 = pts[1] - pts[0];
+    const auto v1 = pts[2] - pts[0];
+    const auto v2 = pts[2] - pts[1];
+
+    const double dotp0 = CGAL::scalar_product(v0,v1);
+    const double angle0 = acos(dotp0 / (sqrt(v0.squared_length()) * sqrt(v1.squared_length())));
+    const double dotp1 = CGAL::scalar_product(-v0, v2);
+    const double angle1 = acos(dotp1 / (sqrt(v0.squared_length()) * sqrt(v2.squared_length())));
+    const double angle2 = CGAL_PI - (angle0 + angle1);
+
+    double curr_min_angle = angle1;
+    if (angle1 < curr_min_angle) curr_min_angle = angle1;
+    if (angle2 < curr_min_angle) curr_min_angle = angle2;
+
+    avg_min_angle += curr_min_angle;
+    if (curr_min_angle < min_angle) min_angle = curr_min_angle;
+  }
+  qavg /= mesh.number_of_faces();
+  avg_min_angle /= mesh.number_of_faces();
+
+  std::cout << "Mesh size: " << mesh.number_of_faces() << std::endl;
+  std::cout << "Average quality: " << qavg << std::endl;
+  std::cout << "Minimum quality: " << qmin << std::endl;
+  std::cout << "Average minimum angle: " << avg_min_angle * 180. / CGAL_PI
+    << " deg" << std::endl;
+  std::cout << "Minimum angle: " << min_angle * 180. / CGAL_PI
+    << " deg" << std::endl;
+
+  return 0;
+}
```

Comment:
add empty line

Reply:
Can you please tell me which line was this aimed at?

Reply:
sure! There should be an empty line at the end of each file and github was complaining about it missing at the end of this file

Reply:
> add empty line

I want to correct. There is no need for an empty line. But each new line for a text file must end with the end-of-line character. That is what was missing here at the end of line 110. Adding that end-of-line will not create an empty line, unless what you call an empty line is a line without any character (without even the end-of-line character).

Reply:
thank you for the precisions